src/tape/ensemble_frame.py

```diff
@@ -688,6 +733,12 @@ def map_partitions(self, func, *args, **kwargs):
         if isinstance(result, self.__class__):
             # If the output of func is another _Frame, let's propagate any metadata.
             return self._propagate_metadata(result)
+        elif isinstance(result, ObjectFrame):
+            result = self._propagate_metadata(result)
+            result.label = "object"  # override the label
+            return result
+        elif isinstance(result, SourceFrame):
+            return self._propagate_metadata(result)
```

Comment:
Can we add a simple test does map_partitions on a SourceFrame?

Am curious if we will need`result.label = SOURCE_FRAME_LABEL` here like with the ObjectFrame

Reply:
Added a test, and looks like it propagates appropriately (for some reason...)